File: Publisher_Subscriber/ex_subscriber.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# Author: hbasriaykol

import rospy
from std_msgs.msg import String


def callback_function(msg):
    ''' This function is callback function that received a message.'''
    rospy.loginfo(msg)


def create_subscriber():
    ''' This function creates a subscriber. '''
    sub = rospy.Subscriber("/a_topic_name", String, callback_function)


if __name__ == '__main__':
    rospy.init_node('a_node_name_3')
    rospy.spin()

File: Publisher_Subscriber/ex_subscriber.cpp
#include<ros/ros.h>
#include<std_msgs/String.h>

void callback_function(const std_msgs::String& msg) 
{
	ROS_INFO(msg.data.c_str());
}

int main(int argc, char **argv)
{
	ros::init(argc, argv, "a_node_name_3");
	ros::NodeHandle nh;

	ros::Subscriber sub = nh.subscribe("/a_topic_name", 1000, callback_function);
	ros::spin();
}
